File: node/argv-parser.js
const defaultParameters = {
    nFrom: 1,
    nTo: 100,
    print: 'series'
}

export const parser = () => {
    const results = defaultParameters

    const argv = process.argv.slice(2).map(item => {
        const [param, value] = item.split('=')

        return { param, value }
    })

    argv.forEach(item => {
        if (results?.[item.param]) {
            results[item.param] = item.value
        }
    })

    // sanitize

    return {
        nFrom: parseInt(results.nFrom),
        nTo: parseInt(results.nTo),
        print: ['series', 'end'].includes(results.print) ? results.print : 'series'
    }
}

File: go/argv_parser.go
package main

import (
	"fizz-buzz/shared"
	"os"
	"strconv"
	"strings"
)

type PrintValue string

var printValues = []string{"series", "end"}

type ArgsType struct {
	nFrom int
	nTo   int
	print string
}

// default parameters, kept global as helper, just in case
var defaultParameters = ArgsType{
	nFrom: 1,
	nTo:   100,
	print: printValues[0],
}

func parser() ArgsType {
	// maps parameters split by '='
	params := make(map[string][]string)

	for _, arg := range os.Args[1:] {
		split := strings.Split(arg, "=")

		params[split[0]] = split[1:]
	}

	// prepare results with default
	results := defaultParameters
	var value []string
	var ok bool

	value, ok = params["nFrom"]
	if ok && len(value) == 1 {
		arg, err := strconv.Atoi(value[0])
		if err != nil {
			results.nFrom = arg
		}
	}

	value, ok = params["nTo"]
	if ok && len(value) == 1 {
		arg, err := strconv.Atoi(value[0])
		if err != nil {
			results.nTo = arg
		}
	}

	value, ok = params["print"]
	if ok && len(value) == 1 {
		arg := value[0]

		if shared.Array_in(printValues, arg) {

			results.print = arg
		}
	}

	return results
}
